synapse/handlers/user_directory.py

```diff
@@ -65,7 +66,7 @@ def __init__(self, hs):
             # we start populating the user directory
             self.clock.call_later(0, self.notify_new_event)
 
-    def search_users(self, user_id, search_term, limit):
+    async def search_users(self, user_id, search_term, limit):
```

Comment:
I believe that making this async and using the await inside of it is OK since `store.search_user_dir` is async, but I'm not 100% sure.

This also makes this file a bit of a weird style with both `defer.inlineCallbacks` and `async` in it.

Reply:
> I believe that making this async and using the await inside of it is OK since store.search_user_dir is async, but I'm not 100% sure.

yup, correct.

> This also makes this file a bit of a weird style with both defer.inlineCallbacks and async in it.

yes, but not a major problem. I'd rather we feel free to incrementally move towards `async` rather than feel held back by having to do entire files at once.

Reply:
Thanks!